packages/ag-charts-enterprise/src/features/error-bar/errorBar.ts

```diff
@@ -88,14 +99,52 @@ export class ErrorBars
         this.cartesianSeries = ctx.series as _ModuleSupport.CartesianSeries<any, any>;
         const { contentGroup } = this.cartesianSeries;
 
-        this.groupNode = new _Scene.Group({ name: `${contentGroup.id}-series-errorBars` });
+        this.groupNode = new _Scene.Group({
+            name: `${contentGroup.id}-series-errorBars`,
+            zIndex: _ModuleSupport.Layers.SERIES_ERRORBAR_ZINDEX,
+            zIndexSubOrder: this.cartesianSeries.getGroupZIndexSubOrder('error-bars'),
+        });
         contentGroup.appendChild(this.groupNode);
         this.selection = _Scene.Selection.select(this.groupNode, () => this.errorBarFactory());
 
-        this.destroyFns.push(this.cartesianSeries.addListener('data-model', (event) => this.onDataProcessed(event)));
+        this.destroyFns.push(
+            this.cartesianSeries.addListener(
+                'processData-prerequest',
+                (event: _ModuleSupport.SeriesPrerequestDataEvent) => this.onPrerequestData(event)
+            )
+        );
+        this.destroyFns.push(
+            this.cartesianSeries.addListener('data-update', (event: _ModuleSupport.SeriesDataUpdateEvent) =>
+                this.onDataUpdate(event)
+            )
+        );
+        this.destroyFns.push(
+            this.cartesianSeries.addListener('visibility-changed', (event: _ModuleSupport.SeriesVisibilityEvent) =>
+                this.onToggleSeriesItem(event)
+            )
+        );
     }
 
-    onDataProcessed(event: {
+    onPrerequestData(event: {
+        props: _ModuleSupport.PropertyDefinition<string>[];
+        isContinuousX: boolean;
+        isContinuousY: boolean;
+    }) {
+        const { cartesianSeries, xLowerKey, xUpperKey, yLowerKey, yUpperKey } = this;
+        const { isContinuousX, isContinuousY } = event;
+        event.props.push(
+            valueProperty(cartesianSeries, yLowerKey, isContinuousY, { id: 'yValue' }),
```

Comment:
Note: I think that I better solution would be to use keys like `yValueLower` and add another series event for when `Series.getDomain()` is called so that `ErrorBars` can add the error values to the domain.

That would be a better solution because right now `ErrorBars` replies on private/internal implementation of the series. This solution would allow the `ErrorBars` class to function independently of the underlying series implementation (including bar series, which doesn't work at the moment).

I plan on doing this in a follow-up PR.